subprojects/packagefiles/msgpackc-cxx/meson.build

```diff
@@ -1,27 +1,36 @@
 project(
-    'msgpackc-cxx', 'cpp',
-    version: '6.1.0',
-    meson_version: '>=0.42.0'
+  'msgpack-cxx', 'cpp',
+  version: '6.1.0',
+  meson_version: '>=0.47.0',
 )
 
 args = []
+
+boost_dep = dependency('boost', required: get_option('boost'))
+if not boost_dep.found()
+  args += ['-DMSGPACK_NO_BOOST']
+endif
+
 if get_option('x3_parse')
-    args += ['-DMSGPACK_USE_X3_PARSE']
+  if not boost_dep.found()
+    error('Boost is required for X3 parse')
+  endif
+  args += ['-DMSGPACK_USE_X3_PARSE']
 endif
+
 if host_machine.cpu_family() == 'sparc'
-    args += ['-DMSGPACK_ZONE_ALIGN=8']
+  args += ['-DMSGPACK_ZONE_ALIGN=8']
 endif
 
-msgpackc_cxx_dep = declare_dependency(
-    compile_args: args,
-    include_directories: include_directories('include'),
-    dependencies: dependency('boost'),
+msgpack_cxx_dep = declare_dependency(
+  compile_args: args,
+  include_directories: include_directories('include'),
+  dependencies: boost_dep,
 )
 
-if meson.version().version_compare('>=0.54.0')
-    meson.override_dependency('msgpackc-cxx', msgpackc_cxx_dep)
-endif
-
 if get_option('tests')
```

Comment:
My preference is to subdir('test') unconditionally and inside if dependencies are missing, pass subdir_done()

Reply:
Sounds reasonable

Reply:
Wait no sorry, I don't get the `subdir_done()` part. The reason why I check for this is that this cannot work if `-DMSGPACK_NO_BOOST` is one of the compile arguments. The only scenario here is that someone uses `boost=disabled` and `tests=true`, which will not work.

Reply:
Simplest way is to do:

```
if get_option('tests') and not boost_dep.found()
  subdir_done() or error()
endif
```

in tests/meson.build. Although it might make more sense to do

```
tboost_dep = dependency('boost', modules: ['unit_test_framework', 'system'], required: get_option('tests'))
```

and match subdir_done() on that.

Reply:
> Simplest way is to do:
> 
> ```
> if get_option('tests') and not boost_dep.found()
>   subdir_done() or error()
> endif
> ```
> 
> in tests/meson.build.

Just so that I get it right, that would be
```meson
# only run tests if required
if not get_option('tests')
  subdir_done()
endif

# ensure boost was not disabled
if not boost_dep.found()
  error('Boost is required for tests')
endif
```

> Although it might make more sense to do
> 
> ```
> tboost_dep = dependency('boost', modules: ['unit_test_framework', 'system'], required: get_option('tests'))
> ```
> 
> and match subdir_done() on that.

No, because that is not enough. We need to check that `boost` has not been disabled (due to the compile definition), so this needs to happen before.

Reply:
Sure. check for boost_dep first and then that.

Reply:
This only really makes sense if `tests` is a feature, not a bool. So I changed that, it now detects automatically if the tests can be build or not. Also allows for the CI to run on Windows.